lib/vanagon/component/dsl.rb

```diff
@@ -424,13 +424,52 @@ def environment(*env) # rubocop:disable Metrics/AbcSize
       def add_preinstall_action(pkg_state, scripts)
         pkg_state = Array(pkg_state)
         scripts = Array(scripts)
-
         if pkg_state.empty? || !(pkg_state - ["install", "upgrade"]).empty?
           raise Vanagon::Error, "#{pkg_state} should be an array containing one or more of ['install', 'upgrade']"
         end
         @component.preinstall_actions << OpenStruct.new(:pkg_state => pkg_state, :scripts => scripts)
       end
 
+      # Adds trigger to run during the install phase of packaging
+      #
+      # @param pkg_state [Array] the state in which the scripts should execute. Can be
+      #   one or multiple of 'install' and 'upgrade'.
+      # @param scripts [Array] the Bourne shell compatible scriptlet(s) to execute
+      # @param pkg [String] the package the trigger will be set in
+      def add_install_triggers(pkg_state, scripts, pkg)
+        pkg_state = Array(pkg_state)
+        scripts = Array(scripts)
+        pkg = String(pkg)
+        if pkg_state.empty? || !(pkg_state - ["install", "upgrade"]).empty?
+          raise Vanagon::Error, "#{pkg_state} should be an array containing one or more of ['install', 'upgrade']"
+        end
+        @component.install_triggers << OpenStruct.new(:pkg_state => pkg_state, :scripts => scripts, :pkg => pkg)
```

Comment:
I think `{ :pkg_state => pkg_state, :scripts => scripts, :pkg => pkg }` is more idiomatic.

Reply:
using OpenStruct is consistent with how we've done this elsewhere in this file

Reply:
ooh

Reply:
Nitpick, don't block on it: I know we do this everywhere else here, but this should probably be using the accessor (`_component`) instead of the instance variable (`@component`).